tests/test_ignore.py

```diff
@@ -0,0 +1,169 @@
+from vulture import core
+
+from . import check, skip_if_not_has_async
+
+
+def check_ignore(code, ignore_names, ignore_decorators, expected):
+    v = core.Vulture(
+            verbose=True, ignore_names=ignore_names,
+            ignore_decorators=ignore_decorators)
+    v.scan(code)
+    check(v.get_unused_code(), expected)
+
+
+def test_var():
+    code = """\
+fio = 1
+fao = 2
+bar = 2
+ftobar = 3
+baz = 10000
+funny = True
+"""
+    check_ignore(code, ['f?o*', 'ba[rz]'], [], ['funny'])
+
+
+def test_function():
+    code = """\
+def foo_one():
+    pass
+def foo_two():
+    pass
+def foo():
+    pass
+def bar():
+    pass
+"""
+    check_ignore(code, ['foo*'], [], ['bar'])
+
+
+@skip_if_not_has_async
+def test_async_function():
+    code = """\
+async def foobar():
+    pass
+async def bar():
+    pass
+"""
+    check_ignore(code, ['foo*'], [], ['bar'])
+
+
+def test_class():
+    code = """\
+class Foo:
+    def __init__(self):
+        pass
```

Comment:
Let's add a second class Bar here that is not ignored. It's always good to have tests for more than once case. You can just use 
```
class Foo:
    pass

class Bar:
    pass
```